tt_metal/impl/dispatch/lock_free_queue.hpp

```diff
@@ -23,6 +23,10 @@ class LockFreeQueue {
         std::atomic<Node*> head;
         std::atomic<Node*> tail;
 
+        std::mutex queue_mutex;
```

Comment:
I have no idea but will raise the question: how long is the lock going to be held?  how often is the lock going to be taken/released?  is this a place of a test and set and busy wait rather than a mutex?

Reply:
The lock will be held only for this part of the `push` call, which writes data into the worker queue slot:

```
while(tail.load()->next == head.load()) {};
tail.load()->data = value;
tail.store(tail.load()->next);
  ```
  
 The critical section is fairly small. We can try with a set and busy lock using atomics to see if we get better perf. Thanks!